source/common/network/raw_buffer_socket.h

```diff
@@ -21,6 +21,8 @@ class RawBufferSocket : public TransportSocket, protected Logger::Loggable<Logge
   IoResult doRead(Buffer::Instance& buffer) override;
   IoResult doWrite(Buffer::Instance& buffer, bool end_stream) override;
   Ssl::ConnectionInfoConstSharedPtr ssl() const override { return nullptr; }
+  // This method must never be called for RawBufferSocket
+  bool startSecureTransport() override { NOT_IMPLEMENTED_GCOVR_EXCL_LINE; }
```

Comment:
This seems potentially a mistake that could be made? Should this should return false?